lib/Runtime/Library/JavascriptBigInt.cpp

```diff
@@ -0,0 +1,205 @@
+    {
+        Assert(charLength >= 0);
+        Assert(pChar != 0);
+        AssertMsg(charLength <= MaxStringLength, "do not support BigInt out of the range");
+
+        const EncodedChar *pCharLimit = pChar + charLength - 1;//'n' at the end
+
+        m_length = 0;
+        m_digits = m_reservedDigitsInit;
+        m_isNegative = isNegative;
+
+        uint32 digitMul = 1;
+        uint32 digitAdd = 0;
+        for (; pChar < pCharLimit; pChar++)
+        {
+            Assert(NumberUtilities::IsDigit(*pChar));
+            if (digitMul == 1000000000)
+            {
+                Assert(MulThenAdd(digitMul, digitAdd));
+                digitMul = 1;
+                digitAdd = 0;
+            }
+            digitMul *= 10;
+            digitAdd = digitAdd * 10 + *pChar - '0';
+        }
+        Assert(1 < digitMul);
+        Assert(MulThenAdd(digitMul, digitAdd));
+    }
+
+    bool JavascriptBigInt::MulThenAdd(uint32 digitMul, uint32 digitAdd)
+    {
+        Assert(digitMul != 0);
+
+        uint32 carryDigit = 0;
+        uint32 *pDigit = m_digits;
+        uint32 *pDigitLimit = pDigit + m_length;
+
+        for (; pDigit < pDigitLimit; pDigit++)
+        {
+            *pDigit = NumberUtilities::MulLu(*pDigit, digitMul, &carryDigit);// return low Digit to digit, hight Digit to carry
+            if (digitAdd > 0)
+            {
+                carryDigit += NumberUtilities::AddLu(pDigit, digitAdd);// add carry to result
+            }
+            digitAdd = carryDigit;
+        }
+        if (0 < digitAdd) // length increase by 1
+        {
+            m_digits[m_length++] = digitAdd;
+        }
+        return true;
+    }
+
+    int JavascriptBigInt::Compare(JavascriptBigInt *pbi)
+    {
+        if (m_isNegative != pbi->m_isNegative)
+        {
+            if (m_isNegative)
+            {
+                return -1;
+            }
+            else
+            {
+                return 1;
+            }
+        }
+
+        uint32 index;
+        int sign = m_isNegative ? -1 : 1;
+
+        if (m_length > pbi->m_length)
+            return 1*sign;
+        if (m_length < pbi->m_length)
+            return -1*sign;
+        if (0 == m_length)
+            return 0;
+
+#pragma prefast(suppress:__WARNING_LOOP_ONLY_EXECUTED_ONCE,"noise")
+        for (index = m_length - 1; m_digits[index] == pbi->m_digits[index]; index--)
+        {
+            if (0 == index)
+                return 0;
+        }
+        Assert(m_digits[index] != pbi->m_digits[index]);
+
+        return sign*((m_digits[index] > pbi->m_digits[index]) ? 1 : -1);
+    }
+
+    bool JavascriptBigInt::LessThan(Var aLeft, Var aRight)
+    {
+        AssertMsg(JavascriptBigInt::Is(aLeft) && JavascriptBigInt::Is(aRight), "BigInt LessThan");
+
+        JavascriptBigInt *leftBigInt = JavascriptBigInt::FromVar(aLeft);
+        JavascriptBigInt *rightBigInt = JavascriptBigInt::FromVar(aRight);
+
+        if (leftBigInt->Compare(rightBigInt) < 0)
```

Comment:
rather than `if (something) return true; return false;`, I prefer `return something;` because it is much shorter